### PR TITLE
Introduce new module reactor-core-micrometer

### PR DESCRIPTION
**Pull request by simonbasle**
groupId: io.projectreactor.observability

 - reactor-metrics-micrometer explicitly depends on micrometer-core
 - it provides an implementation of SignalListener for metrics
 - it supersedes FluxMetrics and Metrics/MicrometerConfiguration in core
 - the above are now deprecated
 - Metrics.isInstrumentationAvailable() will answer false if the new
 module is detected on the classpath

TODO: more tests on the MetricsListener

### CODE REVIEW

**Review by OlegDokuka — APPROVED** [review]

File: reactor-metrics-micrometer/build.gradle [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ * limitations under the License.
+ */
+
+apply plugin: 'biz.aQute.bnd.builder'
+apply plugin: 'java-library'
+
+description = 'Reactor Micrometer Metrics support'
+
+version = "$metricsMicrometerVersion"
+group = "io.projectreactor.observability"
```
Comment (violetagg) [inline reactor-metrics-micrometer/build.gradle:23]: Is this group mandatory for all Reactor projects, or it is intended to be used only by reactor-core?

File: settings.gradle [hunk 2]
```diff
@@ -19,7 +19,7 @@ plugins {
 
 rootProject.name = 'reactor'
 
-include 'benchmarks', 'reactor-core', 'reactor-test', 'reactor-tools'
+include 'benchmarks', 'reactor-core', 'reactor-test', 'reactor-tools', 'reactor-metrics-micrometer'
```
Comment (violetagg) [inline settings.gradle:22]: Is this module name representing all Reactor projects efforts towards metrics with Micrometer or it is related only to reactor-code?

File: reactor-core/src/main/java/reactor/util/Metrics.java [hunk 3]
```diff
@@ -17,62 +17,100 @@
...
+	 * NO-OP).
 	 *
-	 * @return true if the Micrometer instrumentation facade is available
+	 * @return true if the Micrometer instrumentation facade is available, unless the reactor-metrics-micrometer module is also on the classpath
+	 * @deprecated prefer explicit usage of the reactor-metrics-micrometer module. To be removed in 3.6.0 at the earliest.
 	 */
+	@Deprecated
 	public static final boolean isInstrumentationAvailable() {
-		return isMicrometerAvailable;
+		return isMicrometerAvailable && !isMicrometerReactorModuleAvailable;
```
Comment (violetagg) [inline reactor-core/src/main/java/reactor/util/Metrics.java:72]: The purpose of the method is changed!

File: reactor-metrics-micrometer/build.gradle (hunk 1 above)
Comment (simonbasle) [reply reactor-metrics-micrometer/build.gradle]: only by reactor-core. I have changed the package names, module name and bnd configuration to reflect that in the module's name (now `reactor-core-micrometer`)

File: reactor-core/src/main/java/reactor/util/Metrics.java (hunk 3 above)
Comment (simonbasle) [reply reactor-core/src/main/java/reactor/util/Metrics.java]: reverted

**Review by violetagg — APPROVED** [review]

**Pull request merged**